avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1058,12 +1058,44 @@ private Object toConnectData(Schema schema, Object value) {
             arrayVal =
             recordValue.get(ANYTHING_SCHEMA.getField(ANYTHING_SCHEMA_ARRAY_FIELD).pos());
         if (arrayVal != null) {
-          return toConnectData(SchemaBuilder.array(null).build(), arrayVal);
+          // We cannot reuse the logic like we do in other cases because it is not possible to
+          // construct an array schema with a null item schema, but we do not know the schema of the
+          // items.
+          if (!(arrayVal instanceof Collection)) {
```

Comment:
This patch is replacing https://github.com/confluentinc/schema-registry/blob/master/avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java#L1061 with https://github.com/confluentinc/schema-registry/blob/master/avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java#L1138-L1144. The easiest way to implement these and not duplicate code was to determine the implicit schema, then invoke again with that schema and use that conversion code. However, the patch in AK made it impossible to do this for arrays and maps due to them having null schemas in their items.

Regardless, I've just converted the relevant cases throughout to use `Collection` instead.